### PR TITLE
 Keep navigation toolbar location consistent

### PR DESCRIPTION
**Pull request by Kamielvf**
old: https://puu.sh/zwhy8/e05b6d0429.gif
new: https://puu.sh/zweQu/716d58a3ad.gif

Also fixes an issue with the client jumping back to an older size when you resize after opening a panel.

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java [hunk 1]
```diff
@@ -355,15 +354,22 @@ public void windowClosing(WindowEvent e)
...
 		container.add(new ClientPanel(client));
 
-		navContainer = new JPanel();
-		navContainer.setLayout(new BorderLayout(0, 0));
-		navContainer.setMinimumSize(new Dimension(0, 0));
-		navContainer.setMaximumSize(new Dimension(0, Integer.MAX_VALUE));
-		container.add(navContainer);
+		innerNavContainer = new JPanel();
+		innerNavContainer.setLayout(new BorderLayout(0, 0));
+		innerNavContainer.setMinimumSize(new Dimension(0, 0));
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java]: Remove the minimum and maximum size adjustments

File: runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java [hunk 2]
```diff
@@ -355,15 +354,22 @@ public void windowClosing(WindowEvent e)
...
 
 		pluginToolbar = new PluginToolbar(this);
 		container.add(pluginToolbar);
 
+		outerNavContainer = new JPanel();
+		outerNavContainer.setLayout(new BorderLayout(0, 0));
+		outerNavContainer.setMinimumSize(new Dimension(0, 0));
+		outerNavContainer.setMaximumSize(new Dimension(0, Integer.MAX_VALUE));
+		container.add(outerNavContainer);
+
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java]: Remove new line

File: runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java [hunk 3]
```diff
@@ -355,15 +354,22 @@ public void windowClosing(WindowEvent e)
...
+		innerNavContainer.setMinimumSize(new Dimension(0, 0));
+		innerNavContainer.setMaximumSize(new Dimension(0, Integer.MAX_VALUE));
+		container.add(innerNavContainer);
 
 		pluginToolbar = new PluginToolbar(this);
 		container.add(pluginToolbar);
 
+		outerNavContainer = new JPanel();
+		outerNavContainer.setLayout(new BorderLayout(0, 0));
+		outerNavContainer.setMinimumSize(new Dimension(0, 0));
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java]: Remove the minimum size and maximum size adjustments

File: runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java [hunk 4]
```diff
@@ -379,24 +385,28 @@ void expand(PluginPanel panel)
 	{
 		if (pluginPanel != null)
 		{
-			navContainer.remove(0);
+			activeNavContainer.remove(0);
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java]: Remove pluginPanel, not 0

File: runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java [hunk 5]
```diff
@@ -410,19 +420,19 @@ void contract()
 	{
 		boolean wasMinimumWidth = this.getWidth() == (int) this.getMinimumSize().getWidth();
 		pluginPanel.onDeactivate();
-		navContainer.remove(0);
-		navContainer.setMinimumSize(new Dimension(0, 0));
-		navContainer.setMaximumSize(new Dimension(0, Integer.MAX_VALUE));
-		navContainer.revalidate();
+		activeNavContainer.remove(0);
+		activeNavContainer.setMinimumSize(new Dimension(0, 0));
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java]: Remove the minimum and maximum size adjustments

File: runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java [hunk 6]
```diff
@@ -81,10 +81,9 @@
 	private final RuneLite runelite;
 	private final Applet client;
 	private final RuneLiteProperties properties;
-	private JPanel navContainer;
+	private JPanel innerNavContainer, outerNavContainer, activeNavContainer;
```
Comment (devLotto) [inline runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java]:
```
private JPanel innerNavContainer;
private JPanel outerNavContainer;
private JPanel activeNavContainer;
```

**Comment by l2-** [comment]
can the nav button on left or right side be a toggle please? :)

File: runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java [hunk 7]
```diff
@@ -379,24 +385,28 @@ void expand(PluginPanel panel)
...
+			{
+				activeNavContainer = innerNavContainer;
+			}
 		}
 
 		pluginPanel = panel;
-		navContainer.setMinimumSize(new Dimension(PANEL_EXPANDED_WIDTH, 0));
-		navContainer.setMaximumSize(new Dimension(PANEL_EXPANDED_WIDTH, Integer.MAX_VALUE));
+		activeNavContainer.setMinimumSize(new Dimension(PANEL_EXPANDED_WIDTH, 0));
+		activeNavContainer.setMaximumSize(new Dimension(PANEL_EXPANDED_WIDTH, Integer.MAX_VALUE));
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java]: why not just always keep the maximum size at the PANEL_EXPANDED_WIDTH one but change it every time?

File: runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java [hunk 8]
```diff
@@ -410,19 +420,19 @@ void contract()
 	{
 		boolean wasMinimumWidth = this.getWidth() == (int) this.getMinimumSize().getWidth();
 		pluginPanel.onDeactivate();
-		navContainer.remove(0);
-		navContainer.setMinimumSize(new Dimension(0, 0));
-		navContainer.setMaximumSize(new Dimension(0, Integer.MAX_VALUE));
-		navContainer.revalidate();
+		activeNavContainer.remove(0);
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java]: Dont remove 0 but pluginPanel

File: runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java (hunk 7 above)
Comment (Kamielvf) [reply runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java]: It works just fine when setting the maximum size to `PANEL_EXPANDED_WIDTH` when creating `outerNavContainer` and `innerNavContainer` so I guess I should just do that.

**Comment by Kamielvf** [comment]
@l2- I think adding an option for this would be kind of overdoing it

**Comment by deathbeam** [comment]
I agree, too much togglescape

**Comment by l2-** [comment]
too much togglescape but you seriously want to be the first client who breaks UI flow by jamming buttons between the game and the sidepanel

**Comment by deathbeam** [comment]
No, but having the option to change it is worse. Also, it's not like it is decided that this is going to be 100% merged too.

**Comment by Kamielvf** [comment]
Closing as this is no longer relevant.

**Pull request closed without merging**